Game crashes when trying to enter Drassen Airport sector

Load the attached savegame file, unpause the game, then enter the battle. Game will terminate with
the message [ERROR] sgp/SGP.cc: Game has been terminated due to an unrecoverable error: openGameResForReading: VfsFile_open "tilesets/17/t/grass.sti": entity not found

This is with the current master branch and was caused by [this change](https://github.com/ja2-stracciatella/ja2-stracciatella/commit/d08afb162f3685420dde4a24db9e8fd7a4e4b0e9#diff-e0e4d0953818444557b3c1e5003d15e6c8ec33eb58546d1667cca1127c7b9eed): in the case of this crash, res.tilesetID is 17, but there is no file 17/t/grass.sti in Tilesets.slf, so the error message is correct.  Before the change this would have used 0/t/grass.sti, which is working fine. I am using the german v1.05 data files, but that should not make any difference.

[SG10.zip](https://github.com/ja2-stracciatella/ja2-stracciatella/files/5672579/SG10.zip)

--- a/src/game/TileEngine/Overhead_Map.cc
+++ b/src/game/TileEngine/Overhead_Map.cc
@@ -104,7 +104,7 @@ void InitNewOverheadDB(TileSetID const ubTilesetID)
 		}
 		catch (std::exception &e)
 		{
-			SLOGE("%s", e.what());
+			SLOGD(ST::string(e.what()));
 			// Load one we know about
 			vo = AddVideoObjectFromFile(GCM->getTilesetResourceName(GetDefaultTileset(), ST::string("t/") + "grass.sti").c_str());
 		}

--- a/src/game/TileEngine/Overhead_Map.cc
+++ b/src/game/TileEngine/Overhead_Map.cc
@@ -102,10 +102,11 @@ void InitNewOverheadDB(TileSetID const ubTilesetID)
 		{
 			vo = AddVideoObjectFromFile(res.resourceFileName.c_str());
 		}
-		catch (...)
+		catch (std::exception &e)
 		{
+			SLOGE("%s", e.what());
 			// Load one we know about
-			vo = AddVideoObjectFromFile(GCM->getTilesetResourceName(res.tilesetID, ST::string("t/") + "grass.sti").c_str());
+			vo = AddVideoObjectFromFile(GCM->getTilesetResourceName(GetDefaultTileset(), ST::string("t/") + "grass.sti").c_str());
 		}
 
 		gSmTileSurf[i].vo = vo;

--- a/src/game/TileEngine/WorldDef.cc
+++ b/src/game/TileEngine/WorldDef.cc
@@ -206,17 +206,23 @@ void DeinitializeWorld( )
 
 static void AddTileSurface(const ST::string filename, UINT32 const tileType);
 
+TileSetID GetDefaultTileset() {
+	return (gubNumTilesets == JA25_NUM_TILESETS) // If we have the number of tilesets for JA25 useJA25 default, else vanilla default
+		? DEFAULT_JA25_TILESET
+		: GENERIC_1;
+}
+
 TILE_SURFACE_RESOURCE GetAdjustedTilesetResource(TileSetID tilesetID, UINT32 uiTileType, const ST::string filePrefix)
 {
-	if (tilesetID >= gubNumTilesets) throw std::logic_error("invavlid tilesetID");
+	if (tilesetID >= gubNumTilesets) throw std::logic_error("invalid tilesetID");
 
 	ST::string  filename = gTilesets[tilesetID].zTileSurfaceFilenames[uiTileType];
 	if (filename.empty())
 	{
 		// Try loading from default tileset
-		tilesetID = (tilesetID >= DEFAULT_JA25_TILESET || (gubNumTilesets == JA25_NUM_TILESETS && uiTileType == SPECIALTILES))
-			? DEFAULT_JA25_TILESET     //If the map uses JA25 tilesets ( 50 - 69 ) or it is SPECIALTILES (and JA25 tilesets available), use DEFAULT_JA25_TILESET
-			: GENERIC_1                //If the map uses tilesets from Ja2 ( 0 - 49 ), use t0 as the default
+		tilesetID = (gubNumTilesets == JA25_NUM_TILESETS && uiTileType == SPECIALTILES)
+			? DEFAULT_JA25_TILESET     // If the map is SPECIALTILES (and JA25 tilesets available), use DEFAULT_JA25_TILESET
+			: GetDefaultTileset()      // Else use default
 		;
 		filename = gTilesets[tilesetID].zTileSurfaceFilenames[uiTileType];
 	}

--- a/src/game/TileEngine/WorldDef.h
+++ b/src/game/TileEngine/WorldDef.h
@@ -277,6 +277,7 @@ void CompileWorldMovementCosts(void);
 void RecompileLocalMovementCosts( INT16 sCentreGridNo );
 void RecompileLocalMovementCostsFromRadius( INT16 sCentreGridNo, INT8 bRadius );
 
+TileSetID GetDefaultTileset();
 // Tilesets may not provide all tile types.If a tile type is not available in a tileset, we fall back and use the surface in the default tileset.
 TILE_SURFACE_RESOURCE GetAdjustedTilesetResource(TileSetID tilesetID, UINT32 uiTileType, const ST::string filePrefix = "");
 

